risc0/zkvm/lib.rs

```diff
@@ -22,24 +24,27 @@ mod ffi {
 
         type MethodID;
 
-        fn new_method_id(elf_path: &CxxString) -> Result<UniquePtr<MethodID>>;
-        fn write(&self, filename: &CxxString) -> Result<()>;
+        fn make_method_id(elf_path: &CxxString) -> Result<UniquePtr<MethodID>>;
     }
 }
 
+// Can we remove this?
 pub struct MethodID {
     raw: UniquePtr<ffi::MethodID>,
 }
 
 impl MethodID {
     pub fn new(elf_path: &str) -> Result<Self> {
         let_cxx_string!(elf_path = elf_path);
-        let raw = ffi::new_method_id(&elf_path)?;
+        let raw = ffi::make_method_id(&elf_path)?;
         Ok(MethodID { raw })
     }
 
-    pub fn write(&self, filename: &str) -> Result<()> {
-        let_cxx_string!(filename = filename);
-        Ok(self.raw.write(&filename)?)
+    pub unsafe fn to_bytes(&self) -> Result<&[u8]> {
```

Comment:
Why is this unsafe?

Reply:
`slice::from_raw_parts` is unsafe. But if we can make some good safety assumptions then perhaps we can hide the `unsafe` from the `to_bytes` function signature and instead use an internal `unsafe {}` block.

Reply:
After more thought, I don't think we need to convert to a buffer here, since we already have a `MethodID` which has a representation in C++, I think we can just pass a pointer of the raw `ffi::MethodID` down.

Reply:
This is here so we can write the bytes to the methods file though...I think...I'm on mobile and can't really see the whole file.